### ISSUE TITLE
Implement user registration and login/logout

### ISSUE THREAD

**Issue by scottzer0** [AugurProject/augur#132]
In an effort to separate accounts from those management by geth and allow for using a hosted geth node.

@tinybike: update this issue with augur.js implementation details, please.

**Comment by tinybike** [comment]
Register and login methods are implemented in augur.js.  The user specifies a username and password, and a private key (+ derived public key and address) is automatically generated for the user.  The sha256 hash of the password (along with a random 128-bit initialization vector) is used as the aes-256 secret key to encrypt the private key.  The username and the encrypted private key (+ initialization vector) are stored as a key-value pair in the ethereum LevelDB.  (This will need to be changed soon -- possibly to nedb and/or html5 localstorage -- as manual put/get to the ethereum DB is deprecated and slated for removal at some point, as per https://github.com/ethereum/wiki/wiki/JSON-RPC.)

Example use of the `web.register` method:
```
> var username = "tinybike";
> var password = "tinypassword";
> Augur.web.register(username, password)
{ handle: 'tinybike',
  privateKey: <Buffer 55 ec ac 35 98 76 6f b9 05 de a2 13 8c 6d 48 7f 07 9e b1 36 a8 fb eb 5b 93 e2 59 6f 91 4f 35 81>,
  address: '0x09129dcde8e0f510ffb2a715709b69f5c4a42361',
  nonce: 0 }
```
The user is automatically logged in after registering.  When logged in, the `web.account` object is set:
```
> Augur.web.account
{ handle: 'tinybike',
  privateKey: <Buffer 55 ec ac 35 98 76 6f b9 05 de a2 13 8c 6d 48 7f 07 9e b1 36 a8 fb eb 5b 93 e2 59 6f 91 4f 35 81>,
  address: '0x09129dcde8e0f510ffb2a715709b69f5c4a42361',
  nonce: 0 }
```
Logging out unsets the `web.account` object:
```
> Augur.web.logout();
> Augur.web.account
{}
```
Logging in both sets and returns the `web.account` object:
```
> Augur.web.login("tinybike", "tinypassword")
{ handle: 'tinybike',
  privateKey: <Buffer 55 ec ac 35 98 76 6f b9 05 de a2 13 8c 6d 48 7f 07 9e b1 36 a8 fb eb 5b 93 e2 59 6f 91 4f 35 81>,
  address: '0x09129dcde8e0f510ffb2a715709b69f5c4a42361',
  nonce: 0 }
```
Invoking contract methods should be unchanged from normal use (from the client's perspective) -- the client should use the normal Augur API functions (e.g., `Augur.reputationFaucet`), and augur.js defaults to client-side transactions if the user is logged in.

**Comment by tinybike** [comment]
Implemented in cac9c9deba58447e6ac55a8f99d77ed6ca570d6b.
